dhis-2/dhis-test-integration/src/test/java/org/hisp/dhis/tracker/imports/validation/EventImportValidationTest.java

```diff
@@ -207,6 +207,24 @@ void testNonRepeatableProgramStage()
         assertHasOnlyErrors( importReport, ValidationCode.E1039 );
     }
 
+    @Test
+    void testNonRepeatableProgramStageForProgramEvent()
```

Comment:
do we need to assert in two different parts of the test? could we not split it and have each assertion in a separate test?

Reply:
The second assertion depends on the first part to be successfully imported.
To correctly implement this we would need to create a new test class where we have a setup with the correct state for this test case, but for the purpose of this bug fix that is gong to be backported in multiple places, I would prefer to not change too many things